geode-core/src/test/java/org/apache/geode/internal/net/NioSslEngineTest.java

```diff
@@ -265,16 +268,22 @@ public void unwrapWithBufferOverflow() throws Exception {
     spyNioSslEngine.engine = testEngine;
 
     testEngine.addReturnResult(
-        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, netBytes.length, netBytes.length),
+        new SSLEngineResult(BUFFER_OVERFLOW, NEED_UNWRAP, 0, 0), // results in 30,000 byte buffer
```

Comment:
Would it be useful to have incremental assertions for each of these return results being added to the engine?

Reply:
I think you're supposed to do the work and then do assertions.  I'd have to break it into multiple tests to do that.  But why?  If the final result is wrong the intermediate steps failed.  It would only help with debugging, I think.